components/suggestion/command_provider/app_command_parser/app_command_parser.ts

```diff
@@ -1237,33 +1246,92 @@ export class AppCommandParser {
     }
 
     // getUserSuggestions returns a suggestion with `@` if the user has not started typing
-    private getUserSuggestions = (parsed: ParsedCommand): AutocompleteSuggestion[] => {
-        if (parsed.incomplete.trim().length === 0) {
+    private getUserSuggestions = async (parsed: ParsedCommand): Promise<AutocompleteSuggestion[]> => {
+        let input = parsed.incomplete.trim();
+        if (input[0] === '@') {
+            input = input.substring(1);
+        }
+        const {data} = await this.store.dispatch(autocompleteUsersInChannel(input, this.channelID));
+        if (!data) {
             return [{
                 Complete: '',
                 Suggestion: '',
-                Description: parsed.field?.description || '',
-                Hint: parsed.field?.hint || '@username',
-                IconData: parsed.binding?.icon || '',
+                Description: 'No user found',
+                Hint: '',
+                IconData: '',
             }];
         }
+        const usersAutocomplete = data as UserAutocomplete;
+        if (!usersAutocomplete.users.length && !usersAutocomplete.out_of_channel?.length) {
+            return [{
+                Complete: '',
+                Suggestion: '',
+                Description: 'No user found',
+                Hint: '',
+                IconData: '',
+            }];
+        }
+        const items: AutocompleteSuggestion[] = [];
+        usersAutocomplete.users.forEach((u) => {
+            items.push({
+                Complete: '@' + u.username,
+                Suggestion: u.username,
+                Description: u.first_name,
+                Hint: '',
+                IconData: Utils.imageURLForUser(u.id, u.last_picture_update),
+            });
+        });
+        usersAutocomplete.out_of_channel?.forEach((u) => {
+            items.push({
+                Complete: '@' + u.username,
+                Suggestion: u.username,
+                Description: u.first_name,
```

Comment:
Can we use the user's display name preferences here?

Reply:
This is the logic behind the current suggestions. I will apply it (didn't realized it was such an easy copy paste).
```
            itemname = item.username;

            if ((item.first_name || item.last_name) && item.nickname) {
                description = `${Utils.getFullName(item)} (${item.nickname})`;
            } else if (item.nickname) {
                description = `(${item.nickname})`;
            } else if (item.first_name || item.last_name) {
                description = `${Utils.getFullName(item)}`;
            }
```

Reply:
Might as well put it in its own function to share the logic. I assume similar logic will need to be used on mobile as well.